pkg/tracer/tracer.go

```diff
@@ -885,35 +885,37 @@ func (m *FlowFetcher) LookupAndDeleteMap(met *metrics.Metrics) map[ebpf.BpfFlowI
 	// Reiterate on additional metrics
 	var additionalMetrics []ebpf.BpfAdditionalMetrics
 	ids = []ebpf.BpfFlowId{}
-	addtlIterator := m.objects.AdditionalFlowMetrics.Iterate()
-	for addtlIterator.Next(&id, &additionalMetrics) {
-		ids = append(ids, id)
-	}
-
-	countAdditional := 0
-	for i, id := range ids {
-		countAdditional++
-		if err := m.objects.AdditionalFlowMetrics.LookupAndDelete(&id, &additionalMetrics); err != nil {
-			if i == 0 && errors.Is(err, cilium.ErrNotSupported) {
-				log.WithError(err).Warnf("switching to legacy mode")
-				m.lookupAndDeleteSupported = false
-				return m.legacyLookupAndDeleteMap(met)
+	if m.objects.AdditionalFlowMetrics != nil {
```

Comment:
actually, looking at the code, I'm not sure this is what we want here. I rather think that the root cause is missing stuff in paths where bpf objects are being reassigned. In `kernelSpecificLoadAndAssign`.
Also I realize the map object isn't closed in `Close` func

Reply:
hmm look at the stack trace in the PR description we are referrencing null ptr ?

Reply:
I did add handle to kernelspecific and added the missing close while I am here

Reply:
should I keep the sanity check in lookupAndDelete for null ptr ? or not needed anymore ?

Reply:
I removed it will retest

Reply:
thanks ... and sorry for missing all that previously